Makefile

```diff
@@ -15,3 +15,45 @@ apply-copyright:
 fix-licenses: apply-copyright
 
 check-licenses: copyright-check
+
+.PHONY: build-linter-image lint-submodules check-linter-image check-dependencies lint-all mypy
+
+IMAGE := drx-idp:latest
+GIT_COMMIT := $(shell git log -1 --format=%H)
+
+# Build linter image
+build-linter-image:
+	docker build -t $(IMAGE) -f docker/linter/Dockerfile .
+
+check-dependencies:
+	@echo "🎱 Checking dependencies..."
+	pip install -r requirements.txt -r dev-requirements.txt -q
+	@echo "🎱 Dependencies are installed"
+
+check-linter-image:
+	@echo "🎱 Checking if linter image exists..."
+	@if DOCKER_CLI_EXPERIMENTAL=enabled docker manifest inspect $(IMAGE) > /dev/null; then \
+		echo "🪕 image $(IMAGE) already exists. Build has been skipped "; \
+	else \
+		echo "🪇 image $(IMAGE) does not exist. Building..."; \
+		make build-linter-image; \
+	fi
+
+# Run all linters
+lint-all: check-linter-image
+	docker run --rm -v $(PWD):/workspace -w /workspace $(IMAGE) ./linters/run_all.sh
```

Comment:
@jpgomes can you double check the the other team members can run this ok locally? i have docker installed but i'm on an R&D laptop